src/quick-lint-js/fe/parse-statement.cpp

```diff
@@ -2766,15 +2766,51 @@ void Parser::parse_and_visit_switch(Parse_Visitor_Base &v) {
 
   bool keep_going = true;
   bool is_before_first_switch_case = true;
+  Token prev_token;
   Hash_Set<String8_View> cases;
   while (keep_going) {
+    auto is_comment_line = [&]() -> bool {
```

Comment:
@strager the function to find the comment, `is_comment_line` skips all whitespace, which is similar to the lexer function. However, the lexer handles many more cases of characters. Do you think all those cases should be included within `is_comment_line` as well ?

Reply:
Must fix: We should not repeat comment-parsing code like you have done here.

I see two options:
* Call `Lexer`'s code directly instead of reimplementing or copy-pasting it here; or
* Teach the lexer to communicate when a comment was seen before a token.
We should try the latter solution as I think it will be more reliable and useful. I think adding a `bool has_leading_comment;` into the `Token` class should be simple enough for our needs. `Token::has_leading_newline` already exists, and the code for `has_leading_comment` might be very similar.

Reply:
You are right. I went with the suggestion of adding `has_leading_comment` and it worked very well 👍